core/src/main/java/org/jboss/pnc/rex/core/HttpErrorRetryPolicyConfiguration.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021-2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rex.core;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import org.jboss.pnc.rex.common.exceptions.HttpResponseException;
+import org.jboss.pnc.rex.core.config.HttpRetryPolicy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class HttpErrorRetryPolicyConfiguration {
```

Comment:
Can you make this into a native microprofile-config class? Similar to how Backoff425Policy was done. Should be CDI-able + input validation.

Reply:
Looks great!